### PR TITLE
add msbuild 16 support

### PR DESCRIPTION
**Pull request by patricklee2**
support msbuild16
sniff for dotnet3.X
add SCM_USE_MSBUILD16

### CODE REVIEW

File: Build/kudu.ruleset [hunk 1]
```diff
@@ -21,6 +21,7 @@
     <Rule Id="CA1303" Action="None" />
     <Rule Id="CA1305" Action="None" />
     <Rule Id="CA1308" Action="None" />
+    <Rule Id="CA1502" Action="None" />
```
Comment (patricklee2) [inline Build/kudu.ruleset]: cyclomatic complexity of SiteBuilderFactory .CreateBuilder is too high

File: Kudu.Core/Deployment/Generator/SiteBuilderFactory.cs [hunk 2]
```diff
@@ -102,30 +112,30 @@ public ISiteBuilder CreateBuilder(ITracer tracer, ILogger logger, IDeploymentSet
             // shunTODO need to implement this
             VsSolutionProject project = solution.Projects.Where(p => p.IsWap || p.IsWebSite || p.IsAspNetCore || p.IsFunctionApp).FirstOrDefault();
 
-            VsHelper.SniffGlobalJson($"{repositoryRoot}\\global.json");
```
Comment (patricklee2) [inline Kudu.Core/Deployment/Generator/SiteBuilderFactory.cs]: dont need to sniff for global.json

File: Kudu.Core/Infrastructure/AspNetCoreHelper.cs [hunk 3]
```diff
@@ -67,9 +67,8 @@ public static bool IsDotnetCoreFromProjectFile(string projectPath, IEnumerable<G
                     return true;
                 }
 
-                // if ASP.NET CORE 3.X check for <Project Sdk="Microsoft.NET.Sdk.Web"> 
-                return VsHelper.IsAspNetCoreSDK(VsHelper.GetProjectSDK(projectPath)) &&
-                        VsHelper.IsDotNetCore3(VsHelper.GetTargetFramework(projectPath));
+                // check for <Project Sdk="Microsoft.NET.Sdk.Web"> 
+                return VsHelper.IsAspNetCoreSDK(VsHelper.GetProjectSDK(projectPath));
```
Comment (patricklee2) [inline Kudu.Core/Infrastructure/AspNetCoreHelper.cs]: new in 3.0, aspnetcore package is implicitly reference by sdk.web

File: Kudu.Core/Infrastructure/VsHelper.cs [hunk 4]
```diff
@@ -293,11 +239,71 @@ where PathUtilityFactory.Instance.PathsEquals(p.AbsolutePath, targetPath)
...
+                string document = File.ReadAllText(path);
+                return GetTargetFrameworkJsonContents(document);
+            }
+            catch (Exception)
+            {
+                return string.Empty;
+            }
+        }
+
+        public static string GetTargetFrameworkJsonContents(string content)
```
Comment (patricklee2) [inline Kudu.Core/Infrastructure/VsHelper.cs]: https://docs.microsoft.com/en-us/dotnet/core/tools/project-json-to-csproj#one-target-framework

File: Kudu.Core/Infrastructure/VsHelper.cs [hunk 5]
```diff
@@ -293,11 +239,71 @@ where PathUtilityFactory.Instance.PathsEquals(p.AbsolutePath, targetPath)
...
+        }
+
         private static XName GetName(string name)
         {
             return XName.Get(name, "http://schemas.microsoft.com/developer/msbuild/2003");
         }
 
+        public static bool UseMSBuild16()
+        {
+            String var = System.Environment.GetEnvironmentVariable("APPSETTING_"+SettingsKeys.UseMSBuild16);
```
Comment (suwatch) [inline Kudu.Core/Infrastructure/VsHelper.cs]: $"APPSETTING_{}"

File: Build/kudu.ruleset (hunk 1 above)
Comment (suwatch) [inline Build/kudu.ruleset]: remove

**Pull request merged**